[TASK] Move Configuration file description in chapter configuration (#2241)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/Configuration/ConfigurationFiles.rst

.. include:: /Includes.rst.txt

.. _configuration-files:

===================
Configuration files
===================

The configuration files :file:`LocalConfiguration.php` and
:file:`AdditionalConfiguration.php` are located in the directory

> :ref:`public/typo3conf/ <directory-public-typo3conf>` in Composer-based

Reviewer: The reference is not available by now. It is one that is defined in a following PR?
Below the typo3conf ref also.

Author: Yes they are in the follow up. Shall we leave them in?

Reviewer: It would be cleaner. But I am fine with both.

Author: done

> :ref:`public/typo3conf/ <directory-public-typo3conf>` in Composer-based

Author: ```suggestion
:file:`public/typo3conf/` in Composer-based
```

installations. In legacy installations they are located in

> :ref:`typo3conf/ <legacy-directory-typo3conf>`.

Author: ```suggestion
:file:`typo3conf/`.
```

> The most important configurations file is

Reviewer: ```suggestion
The most important configuration file is
```

:file:`LocalConfiguration.php`. It contains local settings of the

> main global PHP array :php:`$GLOBALS['TYPO3_CONF_VARS`], crucial settings

Reviewer: ```suggestion
main global PHP array :php:`$GLOBALS['TYPO3_CONF_VARS']`, crucial settings
```

like database connect credentials are in here. The file is managed by the
:guilabel:`Admin Tools`.

> The file :file:`LocalConfiguration.php` can be enriched by

Author: ```suggestion
The file :file:`LocalConfiguration.php` can be overridden by settings in the file
```

> The file :file:`LocalConfiguration.php` can be enriched by

Reviewer: How can a file be enriched by another file? Maybe better: "The configuration can be adjusted by AdditionalConfiguration" or something similar.

Reviewer 2: The configuration can be overridden by settings in the file AdditionalConfiguration.php. Despite some downsides this is a widely used option as it allows easily different configuration for various environments like production and development without changing any file: the file LocalConfiguration.php is used for production and developers can create their own configuration in AdditionalConfiguration.php.

> The file :file:`LocalConfiguration.php` can be overridden by settings in the file

Reviewer: Maybe better: "The settings in the LocalConfiguration.php file can be overridden in the AdditionalConfiguration.php file."

The current sentence sounds strange: "The file can be overridden by settings in the file ..."

> The file :file:`LocalConfiguration.php` can be overridden by settings in the file
> :file:`AdditionalConfiguration.php` which is never touched by TYPO3

Author: ```suggestion
The settings in the :file:`LocalConfiguration.php`  can be overridden in the 
:file:`AdditionalConfiguration.php` file, which is never touched by TYPO3
```

internal management tools. Be aware that having settings within

> :file:`AdditionalConfiguration.php` may prevent the system from doing

Reviewer: ```suggestion
:file:`AdditionalConfiguration.php` may prevent the system from performing
```

This avoids "doing" used two times in different contextes in the same sentence.
